app/assets/javascripts/oxalis/model/sagas/save_saga.js

```diff
@@ -175,6 +175,20 @@ export function compactUpdateActions(updateActionsBatches: Array<Array<UpdateAct
   });
 
 
+  // This part of the code detects deleted trees.
+  // Instead of sending deleteNode/deleteEdge update actions for all nodes of a deleted tree,
+  // just one deleteTree update action is sufficient for the server to delete the tree.
+  // As the deleteTree update action is already part of the update actions if a tree is deleted,
+  // all corresponding deleteNode/deleteEdge update actions can simply be removed.
+  result = result.map((batch) => {
+    // TODO: Remove the check in map once Flow recognizes that the result of the filter contains only deleteTree update actions
+    const deletedTreeIds = batch.filter(ua => ua.action === "deleteTree").map(ua => ua.action === "deleteTree" && ua.value.id);
```

Comment:
I forgot the name for this behavior, but it's as in Python, where `false && 6` yields `false` and `true && 6` yields `6`. Try it out in the console :)
Of course I'd like to to without it, but as the comment states, it's needed to calm down Flow.